File: EcommerceNashApp.Core/DTOs/Request/ProductRequest.cs
using Microsoft.AspNetCore.Http;

namespace EcommerceNashApp.Core.DTOs.Request
{
    public class ProductRequest
    {
        public string Name { get; set; } = string.Empty;
        public string Description { get; set; } = string.Empty;
        public double Price { get; set; }
        public bool InStock { get; set; } = true;
        public int StockQuantity { get; set; }
        public List<ExistingProductImageRequest> Images { get; set; } = [];
        public List<IFormFile> FormImages { get; set; } = [];
        public List<Guid> CategoryIds { get; set; } = [];
    }
}

File: EcommerceNashApp.Office/src/app/models/requests/productRequest.ts
export interface ExistingProductImageRequest {
  id: string;
  isMain: boolean;
}

export interface ProductRequest {
  name: string;
  description: string;
  price: number;
  inStock: boolean;
  stockQuantity: number;
  images: ExistingProductImageRequest[];
  formImages?: File[];
  categoryIds: string[];
}